### PR TITLE
Match dao

### PR DESCRIPTION
**Pull request by xluxf**
Match entity

### CODE REVIEW

File: dao/pom.xml [hunk 1]
```diff
@@ -46,6 +46,21 @@
...
+        <dependency>
+            <groupId>cz.muni.fi.pa165</groupId>
+            <artifactId>dao</artifactId>
+            <version>1.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>cz.muni.fi.pa165</groupId>
+            <artifactId>dao</artifactId>
+            <version>1.0-SNAPSHOT</version>
+        </dependency>
```
Comment (409754) [inline dao/pom.xml:63]: remove this, it"s dependency on the module itself, even three times :)
it is why the test build failed

File: dao/src/main/java/cz.muni.fi.pa165.soccermanager/dao/MatchDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * @author Filip Lux
+ */
+@Repository
+public class MatchDaoImpl implements MatchDao {
+
+    @PersistenceContext
+    private EntityManager manager;
+
+    @Override
+    public Match fetchById(long MatchId) {
```
Comment (409754) [inline dao/src/main/java/cz.muni.fi.pa165.soccermanager/dao/MatchDaoImpl.java:21]: Variables in java always starts with lower case matchId

File: dao/src/main/java/cz.muni.fi.pa165.soccermanager/dao/PlayerDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package cz.muni.fi.pa165.soccermanager.dao;
```
Comment (409754) [inline dao/src/main/java/cz.muni.fi.pa165.soccermanager/dao/PlayerDaoImpl.java:1]: This should be already in repository

File: dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java [hunk 4]
```diff
@@ -1,4 +1,114 @@
...
+import javax.persistence.*;
+import java.util.Date;
+
 public class Match {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    //homeTeam, awayTeam, place, date, homeTeamGoalsScored, awayTeamGoalsScored
```
Comment (409754) [inline dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java:12]: pls remove help line

File: dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java [hunk 5]
```diff
@@ -1,4 +1,114 @@
...
+
+    @Temporal(TemporalType.DATE)
+    private Date date;
+    private String stadium; // TODO: define enum of stadiums
```
Comment (409754) [inline dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java:16]: resolve todo
maybe enum
maybe each Team could have one its stadium as String .. that could be copied here

File: dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java [hunk 6]
```diff
@@ -1,4 +1,114 @@
...
+        if (this == o) return true;
+        if (!(o instanceof Match)) return false;
+
+        Match match= (Match) o;
+
+        if (getHomeTeam() != match.getHomeTeam()) return false;
+        if (getAwayTeam() != match.getAwayTeam()) return false;
+        if (getDate() != match.getDate()) return false;
+        if (isFinished() != match.isFinished()) return false;
+        if (getId() != null ? !getId().equals(match.getId()) : match.getId() != null) return false;
```
Comment (409754) [inline dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java:80]: We have decided that Id is not a good idea in equals and hashcode :)

File: dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java [hunk 7]
```diff
@@ -1,4 +1,114 @@
...
+    }
+
+    @Override
+    public String toString() {
+        String string = "Match {" +
+                "id=" + getId() +
+                ", date=" + getDate() + '\'' +
+                ", home team=" + getHomeTeam() +
+                ", away team=" + getAwayTeam() +
+                ", stadium=" + getStadium() + '\'' +;
```
Comment (409754) [inline dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java:106]: not sure if java let you keep the final plus..we will see when you fix pom.xml and a new build will be triggered :)

Comment (409754) [reply dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java:106]: btw you can use ?: operator to avoid the if here and just connect strings

**Review by 409754 — CHANGES_REQUESTED** [review]

File: dao/pom.xml (hunk 1 above)
Comment (409754) [reply dao/pom.xml:63]: [INFO] Reactor Summary:
[INFO]
[INFO] Soccer Manager Parent .............................. SUCCESS [ 0.311 s]
[INFO] Persistent layer ................................... FAILURE [ 1.095 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.574 s
[INFO] Finished at: 2017-10-25T16:10:36Z
[INFO] Final Memory: 18M/619M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project dao: Could not resolve dependencies for project cz.muni.fi.pa165:dao:jar:1.0-SNAPSHOT: Failure to find cz.muni.fi.pa165:dao:jar:1.0-SNAPSHOT in https://oss.sonatype.org/content/repositories/snapshots/ was cached in the local repository, resolution will not be reattempted until the update interval of sonatype-snapshots has elapsed or updates are forced -> [Help 1]

File: dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java [hunk 8]
```diff
@@ -1,4 +1,114 @@
 package cz.muni.fi.pa165.soccermanager.entity;
 
+import javax.persistence.*;
+import java.util.Date;
+
 public class Match {
```
Comment (mhamern) [inline dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java:6]: It should be marked with Entity JPA annotation

File: dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java [hunk 9]
```diff
@@ -1,4 +1,114 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    //homeTeam, awayTeam, place, date, homeTeamGoalsScored, awayTeamGoalsScored
+
+    @Temporal(TemporalType.DATE)
+    private Date date;
+    private String stadium; // TODO: define enum of stadiums
+    private Team homeTeam;
```
Comment (mhamern) [inline dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/Match.java:17]: teams should be marked with JPA annotation too (ManyToOne)

**Pull request closed without merging**